.github/workflows/ci.yml

```
@@ -2,30 +2,206 @@ name: CI
+        if: always()
+        run: |
+          echo "### 🧪 Test Results (Python ${{ matrix.python-version }})" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          echo "Tests completed for Python ${{ matrix.python-version }}" >> $GITHUB_STEP_SUMMARY
 
-    - name: Install dependencies
-      run: npm ci
+  build-and-test-node:
+    name: Build and Test (Node.js)
+    runs-on: ubuntu-latest
+    if: hashFiles('package.json') != ''
```

actionlint (1.7.10)

calling function "hashFiles" is not allowed here. "hashFiles" is only available in "jobs.<job_id>.steps.continue-on-error", "jobs.<job_id>.steps.env", "jobs.<job_id>.steps.if", "jobs.<job_id>.steps.name", "jobs.<job_id>.steps.run", "jobs.<job_id>.steps.timeout-minutes", "jobs.<job_id>.steps.with", "jobs.<job_id>.steps.working-directory". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)